packer/plugin/discover.go

```diff
@@ -198,11 +216,62 @@ func (c *Config) discoverSingle(glob string) (map[string]string, error) {
 	return res, nil
 }
 
-func (c *Config) Client(path string) *Client {
+// discoverMultiPlugin takes the description from a multiplugin binary and
+// makes the plugins available to use in Packer. Each plugin found in the
+// binary will be addressable using `${pluginName}-${builderName}` for example.
+// pluginName could be manually set. It usually is a cloud name like amazon.
+// pluginName can be extrapolated from the filename of the binary; so
+// if the "packer-plugin-amazon" binary had an "ebs" builder one could use
+// the "amazon-ebs" builder.
+func (c *Config) discoverMultiPlugin(pluginName, pluginPath string) error {
+	out, err := exec.Command(pluginPath, "describe").Output()
+	if err != nil {
+		return err
+	}
+	var desc pluginsdk.SetDescription
+	if err := json.Unmarshal(out, &desc); err != nil {
+		return err
+	}
+
+	pluginPrefix := pluginName + "-"
+
+	for _, builderName := range desc.Builders {
+		builderName := builderName // copy to avoid pointer overwrite issue
+		c.builders[pluginPrefix+builderName] = func() (packer.Builder, error) {
```

Comment:
At this point, the map is built with keys such as  `amazon_1.0-ebs` because the binary name contains the version, correct? 
Will you deal with the version and names later on in the packer-init command when parsing the required_plugins block?

Reply:
Yes and yes 🙂

Reply:
( to make the PR smaller )